### PR TITLE
[FE] Add changesets

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
This adds [Changesets](https://github.com/changesets/changesets) to our repo to help manage releases for our TypeScript libraries. This should help us quite a bit, especially as the number of packages increases.

The general workflow is this:
- When making a change to a released package (either a single package, or multiple), run `pnpm changeset`. An interactive dialog will prompt you to select the packages you wish to have updated versions, and for a changelog message.
- After merging your PR to main, an automated process will create a [snapshot release](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md) of the updates in main, and publish under the `experimental` tag. This will let external partners use the newer npm packages without requiring them to build and link the packages locally.
- After a new changeset is merged, the action will automatically create a PR to update the versions and changelogs based on the changesets. This allows package version updates to be batched and released together. (example PR here: https://github.com/hayes/pothos/pull/559)
- When we're ready, we can merge the automated PR, after which the updated packages will automatically be deployed to the npm registry.
  - This step is not currently implemented due to some coordination requirements between repos. I'll implement this after this PR lands. For now, we can also manually release after merging the automated PR.

Corresponding PR: https://github.com/MystenLabs/sui-operations/pull/306/

### CODE REVIEW

File: .changeset/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changesets
+
+Hello and welcome! This folder has been automatically generated by `@changesets/cli`, a build tool that works
```
Comment (randall-Mysten) [inline .changeset/README.md:3]:
```suggestion
Hello and welcome! This folder was automatically generated by `@changesets/cli`, a build tool that works
```

File: .changeset/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+with multi-package repos, or single-package repos to help you version and publish your code. You can
+find the full documentation for it [in our repository](https://github.com/changesets/changesets)
+
+We have a quick list of common questions to get you started engaging with this project in
```
Comment (randall-Mysten) [inline .changeset/README.md:7]:
```suggestion
You can view a list of [common questions](https://github.com/changesets/changesets/blob/main/docs/common-questions.md) to help you get started with changesets. 
```

File: .changeset/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+[our documentation](https://github.com/changesets/changesets/blob/main/docs/common-questions.md)
```
Comment (randall-Mysten) [inline .changeset/README.md:8]:
```suggestion

```

**Review by randall-Mysten — COMMENTED** [review]
The strings are worded as though changesets is part of Sui. Reworded to make it 3rd party.

**Review by randall-Mysten — APPROVED** [review]

**Review by bhenhsi — APPROVED** [review]

**Comment by 666lcz** [comment]
> After merging your PR to main, an automated process will create a [snapshot release](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md?rgh-link-date=2022-08-22T17%3A24%3A57Z) of the updates in main, and publish under the experimental tag.

A n00b question: Say if I merged a PR without creating a changeset, this process will not run until the next PR with a changeset is merged right?

**Review by 666lcz — APPROVED** [review]
Looks great!

**Comment by Jordan-Mysten** [comment]
> A n00b question: Say if I merged a PR without creating a changeset, this process will not run until the next PR with a changeset is merged right?

Yeah if you don't create a changeset, then it won't do anything.

**Pull request merged**